sycl/source/platform.cpp

```diff
@@ -87,7 +87,7 @@ platform::get_backend_info() const {
 
 #undef __SYCL_PARAM_TRAITS_SPEC
 
-context platform::ext_oneapi_get_default_context() const {
+context platform::khr_get_default_context() const {
   if (!detail::SYCLConfig<detail::SYCL_ENABLE_DEFAULT_CONTEXTS>::get())
     throw std::runtime_error("SYCL default contexts are not enabled");
```

Comment:
`khr_get_default_context` does not document any exceptions being thrown. If we want this functionality to be "optional", then we need to make sure that its enabled by default to satisfy the extension and explicitly document that opt-out mechanism leads to the following exception being thrown.

An alternative is to keep disabling mechanism to `oneapi` extension and make `khr` always on.

Reply:
I think we can just delete this exception, as it's also not documented by [sycl_ext_oneapi_default_context](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_default_context.asciidoc)?

Reply:
I updated the new function - b0ad586, no it doesn't throw an exception.